awswrangler/distributed/_pool.py

```diff
@@ -2,22 +2,19 @@
 
 import itertools
 import logging
-from typing import Any, Callable, List, Optional, Union
+from typing import Any, Callable, List
 
 import boto3
-from ray.util.multiprocessing import Pool
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
 class _RayPoolExecutor:
-    def __init__(self, processes: Optional[Union[bool, int]] = None):
-        self._exec: Pool = Pool(processes=None if isinstance(processes, bool) else processes)
+    def __init__(self) -> None:
+        pass
 
     def map(self, func: Callable[..., List[str]], _: boto3.Session, *args: Any) -> List[Any]:
-        futures = []
+        """Map func and return ray futures."""
         _logger.debug("Ray map: %s", func)
-        # Discard boto3.Session object & call the fn asynchronously
-        for arg in zip(itertools.repeat(None), *args):
-            futures.append(self._exec.apply_async(func, arg))
-        return [f.get() for f in futures]
+        # Discard boto3.Session object & return futures
+        return list(func(*arg) for arg in zip(itertools.repeat(None), *args))
```

Comment:
So you're essentially using naive `.remote()` implementation instead of the pool just wrapping it in the same interface... 

Considering the issues we've had with the Pool so far I think this might be the way to go, the only thing is that I wanted pool to handle the results & avoid having to call `ray.get if config.distributed` outside thus moving as much ray machinery as possible into this class for the benefit of SOC. 

This would require a feeding all tasks into the pool at the same time similar to what was done [here](https://github.com/awslabs/aws-data-wrangler/blob/feat-3.0/distributed-s3-select-concurrent-multipath/awswrangler/s3/_select.py#L145). Thoughts?

Reply:
Agreed - as I said I am not a fan of having to call a wrapper to just do a `ray.get` and also to flatten a list, just couldn't see a way around it for this use case where we have two ray remote methods (path and scan range)